azext_iot/common/_azure.py

```diff
@@ -13,9 +13,10 @@
 
 def _parse_connection_string(cs, validate=None, cstring_type='entity'):
     decomposed = validate_key_value_pairs(cs)
+    decomposed_lower = dict((k.lower(), v) for k, v in decomposed.items())
     if validate:
         for k in validate:
-            if not decomposed.get(k):
+            if not decomposed.get(k) and not decomposed_lower.get(k.lower()):
```

Comment:
Consider this:  `if not any([decomposed.get(k), decomposed_lower.get(k.lower())]):`

Reply:
fixed